packages/office-ui-fabric-react/src/components/SpinButton/SpinButton.types.ts

```diff
@@ -117,6 +117,12 @@ export interface ISpinButtonProps {
    */
   onDecrement?: (value: string) => string | void;
 
+  /**
+   * This callback is triggered after SpinButton normalized and validated the user input value.
+   * @param value - The normalized and valid value in the input box.
+   */
+  onValidValueUpdated?: (value: string) => void;
```

Comment:
>onValidValueUpdated?: (value: string) => void; [](start = 2, length = 46)

Should there be some information on how this should be used taking into account the onIncrement, onDecrement, and onValidate props? Also, how do those props and this one work together? If they don't work together in a sensible way there should be a warn mutually exclusive telling creators not to combine them (potentially as well as something in the documentation here)

Reply:
They are not really related. They work together great as the way they work. The onIncrement/onDecrement/onValidate is to **inject** some customized logic, the onValidValueUpdated is to **listen** on something happens. I think the doc here is completed, mention something unrelated is not wise.

Reply:
@JasonGore I am fine with any naming part. IMO, one difference of `onValidValueUpdated` and `onValidated` is do we want to trigger the callback after validated **with the same value**. Again, I am fine with both.

I totally agree that the name exposes the architectural issue of SpinButton. I could like to help if we have a clear direction on goal. Could you please like to provide your ideas/suggestions to my questions [here](https://github.com/OfficeDev/office-ui-fabric-react/issues/5326#issuecomment-452992999)?